packages/swapper/src/swappers/ChainflipSwapper/getChainflipTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,276 @@
+    affiliateBps: commissionBps,
+  } = input
+
+  if (!isSupportedChainId(sellAsset.chainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (!isSupportedChainId(buyAsset.chainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (!isSupportedAsset(sellAsset.chainId, sellAsset.symbol)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${sellAsset.name}' on chainId '${sellAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+        details: { chainId: sellAsset.chainId, assetId: sellAsset.assetId },
+      }),
+    )
+  }
+
+  if (!isSupportedAsset(buyAsset.chainId, buyAsset.symbol)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${buyAsset.name}' on chainId '${buyAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+        details: { chainId: buyAsset.chainId, assetId: buyAsset.assetId },
+      }),
+    )
+  }
+
+  const sellChainflipChainKey = `${sellAsset.symbol.toLowerCase()}.${chainIdToChainflipNetwork[sellAsset.chainId as KnownChainIds]}`;
+  const buyChainflipChainKey = `${buyAsset.symbol.toLowerCase()}.${chainIdToChainflipNetwork[buyAsset.chainId as KnownChainIds]}`;
+
+  const brokerUrl = deps.config.REACT_APP_CHAINFLIP_API_URL
+  const apiKey = deps.config.REACT_APP_CHAINFLIP_API_KEY;
+  
+  // Subtract the 0.05% BaaS fee to end up at the final displayed commissionBps
+  let serviceCommission = parseInt(commissionBps) - 5;
+  if (serviceCommission < 0)
+    serviceCommission = 0;
+  
+  const maybeQuoteResponse = await chainflipService.get<ChainflipBaasQuoteQuote[]>(
+    `${brokerUrl}/quotes-native?apiKey=${apiKey}&sourceAsset=${sellChainflipChainKey}&destinationAsset=${buyChainflipChainKey}&amount=${sellAmount}&commissionBps=${serviceCommission}`,
+  );
+
+  // TODO: Throw SellAmountBelowMinimum if that's the cause (parse error response)
+  if (maybeQuoteResponse.isErr()) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'Quote request failed',
+        code: TradeQuoteError.NoRouteFound,
+      }),
+    )
+  }
+
+  const { data: quoteResponse } = maybeQuoteResponse.unwrap()
+  
+  const getGasFee = async () => {
+    const { chainNamespace } = fromAssetId(sellAsset.assetId);
+
+    switch (chainNamespace) {
+      case CHAIN_NAMESPACE.Evm: {
+        const sellAdapter = deps.assertGetEvmChainAdapter(sellAsset.chainId);
+        return await getEvmTxFees({
+          adapter: sellAdapter,
+          supportsEIP1559: (input as GetEvmTradeQuoteInput).supportsEIP1559,
+        });
+      }
+
+      case CHAIN_NAMESPACE.Utxo: {
+        // TODO: Figure out BTC gas calc
```

Comment:
For this one, since we're talking regular sends you can copypaste the logic from sends - the current app logic assumes wallet connected so that makes things pretty straightforward to calculate! 
```typescript
  const getFeeDataInput: GetFeeDataInput<UtxoChainId> = {
        to,
        value,
        chainSpecific: { from, pubkey: account },
        sendMax,
      }
      return adapter.getFeeData(getFeeDataInput)
```

Reply:
I am not very familiar with utxo chains, if it's just to get a quote, what should I use for `account`? A hardcoded bitcoin address?

Reply:
`pubkey` should be `(input as GetUtxoTradeQuoteInput).xpub` here, this is the xpub (we call it xpub for the sake of simplicity but this is really xpub/ypub/zpub or other UTXO chains equivalents e.g dgub for DOGE) for the selected `accountId` you're getting a quote for .

And as far as `from` is concerned, if Chainfilp is happy with sends from any addy (which I assume it is), then you can safely omit it - keep reading for context if you're curious why this exists.

The reason it's there is because THORFi (lending, savers etc... but not swapper) doesn't play nicely with extended public keys but rather requires an *address*, it needs the Tx to consume one or many UTXOs from a single address only, so it can use said address as an identifier. This means the Tx we send (and the one we estimate too by extension, which is what `from` does here) should take that into account and filter out UTXOs used as input to be only the ones of said chosen *address*!

Reply:
Thanks. One more question on this, at the point of the quote I don't know the `to` addy yet. What should I do with that? Is there a magic hardcoded addy to use?

Reply:
Does Chainflip support custom receive addies @CumpsD 
Either way, `to` should be `receiveAddress` - which will either be derived from wallet or custom! `receiveAddress` should already be passed down as input from web, so there shouldn't be any need to derive anything new here.

Reply:
Well, you are actually sending to a deposit address, which will then swap it for you. And that deposit address is generated at time of swap, not at time of quote

Reply:
Oh I see. Then `to` should be the deposit address, similar to e.g in THOR you swap sending to an inbound address. If we don't know it at the time of the quote, this may be one of many chicken and eggs scenarios we're currently reworking our flow to avoid. 
For the time being, better to assume 0 fees then, if we can't get them. We can always bring best effort estimations later on, just using a dummy address / bytes size / price per byte at quote time later on when we're done with this refactor, but that's not a now thing 🙏🏽 (cc @kaladinlight)